### ISSUE TITLE
`for ( async of` should be a syntax error

### ISSUE THREAD

**Issue by Zzzen** [microsoft/TypeScript#43879]
# Bug Report

### 🔎 Search Terms

for async of syntax error

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.2.3#code/G4QwTgBCDOCeB2BjA3AWAFADMD2kAUMCiE2mEA2gCwBMAugJQQDeAvhotvNNgDYCmAOh7YA5gThJ6yIA)

### 💻 Code

```ts
var async;
for (async of [42]) {}
console.log(async);
```

### 🙁 Actual behavior

no error reported.

### 🙂 Expected behavior

The compiler should throw a syntax error: `Uncaught SyntaxError: The left-hand side of a for-of loop may not be 'async'`.

context: https://github.com/tc39/ecma262/issues/2034
